packages/utils/src/fastHooks.ts

```diff
@@ -0,0 +1,72 @@
+import * as React from 'react';
+import { useRefWithInit } from './useRefWithInit';
+
+export type Instance = {
+  didInitialize: boolean;
+};
+
+type HookType = {
+  before: (instance: any) => void;
+  after: (instance: any) => void;
+};
+
+const hooks: HookType[] = [];
```

Comment:
Singleton in the utils package here, but I guess it's not harmful, even if utils package gets duplicated, right?